plugins/nordic-hid/fu-nordic-hid-cfg-channel.c

```diff
@@ -419,14 +423,75 @@ fu_nordic_hid_cfg_channel_cmd_receive(FuNordicHidCfgChannel *self,
 }
 
 static gboolean
-fu_nordic_hid_cfg_channel_add_peers(FuNordicHidCfgChannel *self, GError **error)
+is_cached_peer_connected(guint8 peer_cache_val)
+{
+	return (peer_cache_val % 2) != 0;
+}
+
+static void
+fu_nordic_hid_cfg_channel_add_peer(FuNordicHidCfgChannel *self, guint8 peer_id)
+{
+	g_autoptr(GError) error_peer = NULL;
+	g_autoptr(FuNordicHidCfgChannel) peer = NULL;
+
+	peer = fu_nordic_hid_cfg_channel_new(peer_id, self);
+
+	/* ensure that the general quirk content for Nordic HID devices is applied */
+	fu_device_add_instance_id(FU_DEVICE(peer), "HIDRAW\\VEN_1915");
+
+	if (!fu_device_setup(FU_DEVICE(peer), &error_peer)) {
+		g_debug("failed to discover peer 0x%02x: %s", peer_id, error_peer->message);
+	} else {
```

Comment:
maybe return from the error and remove the else clause?

Reply:
sure